resource/resource.go

```diff
@@ -180,7 +186,12 @@ func (r *Resource) Out(
 		context = fmt.Sprintf("%s/%s", prefix, context)
 	}
 
-	status := github.NewStatus(github.API, token, owner, repo, context)
+	apiServer := fmt.Sprintf("https://api.%s", github.HOST) // default
+	if hostname, ok := source["github_host"].(string); ok {
+		apiServer = fmt.Sprintf("https://api.%s", hostname)
```

Comment:
If someone uses the `source.github_host` param and sets it to say `github.company.com`, this code assumes that their API will be hosted at `api.github.company.com`. This is how most GHE instances I have encountered are set up, and that's how public GitHub is also set up.  I didn't want to introduce two new source params, one for the API, and another for the host.

Should a target GHE's API be hosted on a completely different domain, this code will break. I am open to suggestions on how this can be improved.